apollo-federation/src/sources/connect/json_selection/apply_to.rs

```diff
@@ -298,18 +322,16 @@ impl ApplyTo for PathSelection {
                             json_type_name(data),
                         )
                         .as_str(),
-                        input_path,
+                        input_path_with_key.to_vec(),
                     ));
-                    input_path.pop();
                     return None;
                 }
 
-                let result = if let Some(child) = match key {
+                if let Some(child) = match key {
                     Key::Field(name) => data.get(name),
                     Key::Quoted(name) => data.get(name),
-                    Key::Index(index) => data.get(index),
                 } {
```

Comment:
Could use `data.get(key.as_str())` if a previous suggestion is accepted

Reply:
https://github.com/apollographql/router/pull/5762/commits/489cf5575a63a18f93c79b0a96d7deb28eb4c651